# Exploratory Data Analysis

In [7]:
import pandas as pd

# import dataset
total_data = pd.read_csv("../data/raw/diabetes_data.csv")

### Exploration and data cleaning

We'll try to understand the data and its main features.

In [8]:
# Obtain the dimensions of the dataset
total_data.shape

(3140, 108)

In [9]:
# Obtain information about data types and non-null values
total_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3140 entries, 0 to 3139
Columns: 108 entries, fips to Urban_rural_code
dtypes: float64(61), int64(45), object(2)
memory usage: 2.6+ MB


From here we can deduct that we're dealing with a dataset of 3140  rows and 108 columns

Let's do a quick search of duplicates.

In [10]:
total_data.duplicated().sum()

0

The model seems to have too many columns to make a reasonable EDA, so let's reduce the analysis.

In [12]:
total_data.to_csv("../data/interim/diabetes_data.csv", index=False)